core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -126,13 +126,12 @@
     private final DynamicFiltersFetcher dynamicFiltersFetcher;
 
     private final DynamicFiltersCollector outboundDynamicFiltersCollector;
-    @GuardedBy("this")
     // The version of dynamic filters that has been successfully sent to the worker
-    private long sentDynamicFiltersVersion = INITIAL_DYNAMIC_FILTERS_VERSION;
+    private final AtomicLong sentDynamicFiltersVersion = new AtomicLong(INITIAL_DYNAMIC_FILTERS_VERSION);
 
-    @GuardedBy("this")
+    @GuardedBy("httpClient")
     private Future<?> currentRequest;
-    @GuardedBy("this")
+    @GuardedBy("httpClient")
     private long currentRequestStartNanos;
```

Comment:
I think `currentRequestStartNanos` can be made a member variable of `UpdateResponseHandler` and removed from here, then we don't need locking for it. `TaskStatusResponseCallback` and `DynamicFiltersResponseCallback` already do that

Reply:
Maybe, but it doesn't really simplify the code here as this is always updated together with `currentRequest`